Open.Nat/Discovery/Searcher.cs

```diff
@@ -61,16 +61,19 @@ public Task<IEnumerable<NatDevice>> Search(CancellationToken cancelationToken)
 #else
 		public async Task<IEnumerable<NatDevice>> Search(CancellationToken cancelationToken)
 		{
-			await Task.Factory.StartNew(_ =>
-				{
-					NatDiscoverer.TraceSource.LogInfo("Searching for: {0}", GetType().Name);
-					while (!cancelationToken.IsCancellationRequested)
-					{
-						Discover(cancelationToken);
-						Receive(cancelationToken);
-					}
-					CloseUdpClients();
-				}, null, cancelationToken);
+            if (!cancelationToken.IsCancellationRequested)
```

Comment:
It should be more precise/correct to simply invoke the `cancellationToken.ThrowIfCancellationRequested();` and not only here but also in the NET35 version of this same method.

Reply:
Good call!